projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -174,14 +174,15 @@ public function get_headline() {
 	 * @returns string
 	 */
 	public function filter_add_target_to_dom( $content ) {
-		if ( has_block( 'jetpack/related-posts' ) ) {
+		if ( has_block( 'jetpack/related-posts' ) || Blocks::is_fse_theme() ) {
```

Comment:
To avoid mid-deploy fatals, should we add a `function_exists` check before calling this method? Just thinking about when this change lands, during deployment this file could be updated before the `class-blocks.php` file is updated since that change is deployed as part of a separate package (I think!)

Reply:
I believe this will be okay because of the [`use Automattic\Jetpack\Blocks;` ](https://github.com/Automattic/jetpack/pull/19615/files#diff-9105b068cf482dd9007ac0a2ff0985bf13dac7d398fb802678224448b5897841L4). (The constructor in this class is already using `jetpack_register_block` from the Blocks package.)

Reply:
I was wondering if this one might be slightly trickier. In the generated wpcom diff, it doesn't appear to include the change to `class-blocks.php` (I assume because it's part of the blocks package?). I'm not _quite_ sure the process for doing package updates, but Jeremy handled it for me in https://github.com/Automattic/jetpack/pull/19197. We might either need to manually add the change to the wpcom diff, or add the check so that the two changes can be deployed separately?

Reply:
Ah sorry, I see what you mean -- since the diff already needs manual changes anyway, I've gone ahead and added the change to `class-blocks.php` to the diff as well. @jeherve Does that look like the right approach to you?

Reply:
We'll actually do things in 3 parts, once that PR is merged:

1. We'll release a new version of the Blocks package. This is done by running `tools/changelogger-release.sh packages/blocks` while on `master`, and then committing the output in a new branch, and opening a PR. Once that PR gets merged, a new version of the Blocks package will automatically become available in Packagist.
2. On WordPress.com, update the version of Blocks. Here is an example from the last time I did it: D59174-code. That will bring the new method to WordPress.com simple.
3. Commit D60508-code to start using the new method.

That should work fine in this case. This turned out to be a bit more involved this time around; in the future I think it would make sense, whenever making changes to the Blocks package (or any other package), to do the changes to the package in its own PR, and then start using those changes in a separate PR, to keep PRs small.

Reply:
Thanks for explaining that @jeherve -- I'll definitely break things out in the future. 

Testing my understanding: I've added the wpcom manual change back to the diff after rebasing (to prevent early asset enqueueing now that this PR has changed how that is handled), but I have _not_ added the `class-blocks.php` changes, since that will be a separate diff (your Step 2).